core/src/main/java/org/apache/carbondata/core/keygenerator/directdictionary/timestamp/TimeStampDirectDictionaryGenerator.java

```diff
@@ -39,37 +39,32 @@
  */
 public class TimeStampDirectDictionaryGenerator implements DirectDictionaryGenerator {
 
-  private TimeStampDirectDictionaryGenerator() {
+  private ThreadLocal<SimpleDateFormat> threadLocal = new ThreadLocal<>();
 
-  }
-
-  public static TimeStampDirectDictionaryGenerator instance =
-      new TimeStampDirectDictionaryGenerator();
+  private String dateFormat;
 
   /**
    * The value of 1 unit of the SECOND, MINUTE, HOUR, or DAY in millis.
    */
-  public static final long granularityFactor;
+  public  long granularityFactor;
   /**
    * The date timestamp to be considered as start date for calculating the timestamp
    * java counts the number of milliseconds from  start of "January 1, 1970", this property is
    * customized the start of position. for example "January 1, 2000"
    */
-  public static final long cutOffTimeStamp;
+  public  long cutOffTimeStamp;
   /**
    * Logger instance
    */
+
   private static final LogService LOGGER =
-      LogServiceFactory.getLogService(TimeStampDirectDictionaryGenerator.class.getName());
+          LogServiceFactory.getLogService(TimeStampDirectDictionaryGenerator.class.getName());
 
-  /**
-   * initialization block for granularityFactor and cutOffTimeStamp
-   */
-  static {
+  public TimeStampDirectDictionaryGenerator(String dateFormat) {
```

Comment:
please keep default dateformat TimeStampDirectDictionaryGenerator() construct method, If DataLoading command didn't provide dateformat option for some column, we can use none-parameter construct method.